File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        
        for str in strs:
            key = ''.join(sorted(list(str)))
            if key in dic:
                dic[key].append(str)
            else:
                dic[key] = [str]
        
        return dic.values()

File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
    	HashMap<String, List<String>> map = new HashMap();
    	
    	for (String str : strs) {
    		String key = Stream.of(str.split(""))
    		    .sorted()
    		    .collect(Collectors.joining());

    		if (! map.containsKey(key)) {
    			map.put(key, new ArrayList<>());
    		}
    		
    		map.get(key).add(str);
    	}
    	
        return new ArrayList<List<String>>(map.values());
    }
}